SonarQube.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```diff
@@ -184,6 +184,17 @@
       <SonarQubeAnalysisFiles Remove="@(SonarQubeAnalysisFiles)" Condition=" %(SonarQubeAnalysisFiles.SonarQubeExclude) == 'true' OR %(SonarQubeAnalysisFiles.AutoGen) == 'true' " />
     </ItemGroup>
 
+    <!-- Exclude items under the intermediate output folder e.g. XAML files that are auto-generated on build -->
+    <PropertyGroup>
+      <!-- Calculate the full path to the base intermediate folder -->
+      <SQFullIntermediateOutputPath>$([System.IO.Path]::Combine($(MSBuildProjectDirectory),$(BaseIntermediateOutputPath)))</SQFullIntermediateOutputPath>
+    </PropertyGroup>
+    <ItemGroup>
+      <SQIntermediateFiles Include="$(SQFullIntermediateOutputPath)**\*.*" /> <!-- Remove based on absolute path -->
+      <SQIntermediateFiles Include="$(BaseIntermediateOutputPath)**\*.*" /> <!-- Remove based on relative path -->
+      <SonarQubeAnalysisFiles Remove="@(SQIntermediateFiles)" />
+    </ItemGroup>
```

Comment:
The simplest approach is to create a new item group that uses wildcards to discover all of the files under the intermediate folder and to exclude them from analysis. The wildcards will pick up files that wouldn't be analysed anyway (e.g. *.dll, *.pdb), but that doesn't matter.
I decided to exclude everything under the base intermediate path (i.e. "obj") rather the specific flavour of project being built (e.g. "obj\debug), just in case there are any tools that generate files to the base dir.
I also decided to exclude using both absolute and relative paths so it doesn't matter how the generated files are referenced.